addons/website_sale/models/product.py

```diff
@@ -430,6 +430,14 @@ class Product(models.Model):
 
     product_variant_image_ids = fields.One2many('product.image', 'product_variant_id', string="Extra Variant Images")
 
+    website_url = fields.Char('Website URL', compute='_compute_product_website_url', help='The full URL to access the document through the website.')
+
+    @api.multi
+    def _compute_product_website_url(self):
```

Comment:
You need to define dependencies to your compute function. Try testing your branch before sending it to review, here, if you change the attribute of one your variant, the URL will not be updated.